delivery_carrier_service_level/models/carrier_service_level.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2013-2020 Therp BV <https://therp.nl>.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ServiceLevel(models.Model):
+    _name = "carrier.service.level"
+    _description = "Carrier service levels"
+    _order = "carrier_id, code"
+
+    carrier_id = fields.Many2one("delivery.carrier", string="Carrier")
```

Comment:
Make carrier_id and code required as well. Sorry for not remarking this before...